vm/isolated/isolated.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package isolated
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/google/syzkaller/pkg/config"
+	. "github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/vm/vmimpl"
+)
+
+func init() {
+	vmimpl.Register("isolated", ctor)
+}
+
+type Config struct {
+	Target_Addr string // target machine
+	Target_Dir  string // directory to copy/run on target
+}
+
+type Pool struct {
+	env *vmimpl.Env
+	cfg *Config
+}
+
+type instance struct {
+	cfg    *Config
+	closed chan bool
+	debug  bool
+	sshkey string
+	port   int
+}
+
+func ctor(env *vmimpl.Env) (vmimpl.Pool, error) {
+	cfg := &Config{}
+	if err := config.LoadData(env.Config, cfg); err != nil {
+		return nil, err
+	}
+	if cfg.Target_Addr == "" {
+		return nil, fmt.Errorf("config param targethost is empty")
+	}
+	if cfg.Target_Dir == "" {
+		return nil, fmt.Errorf("config param targetdir is empty")
+	}
+	// sshkey is optional
+	if env.Sshkey != "" && !osutil.IsExist(env.Sshkey) {
+		return nil, fmt.Errorf("ssh key '%v' does not exist", env.Sshkey)
+	}
+	env.Target_WorkDir = cfg.Target_Dir
+	pool := &Pool{
+		cfg: cfg,
+		env: env,
+	}
+	return pool, nil
+}
+
+func (pool *Pool) Count() int {
+	return 1 // no support for multiple devices yet
```

Comment:
Do you mind resolving this now?
I would not ask, but it looks almost trivial. Looking at adb VM, this seems to be all you need to do:

```go
type Config struct {
	Devices []string // list of adb device IDs to use

...

func ctor(env *vmimpl.Env) (vmimpl.Pool, error) {
...
	if env.Debug {
		cfg.Devices = cfg.Devices[:1]
	}

...

func (pool *Pool) Create(workdir string, index int) (vmimpl.Instance, error) {
	inst := &instance{
		device: pool.cfg.Devices[index],
```

Will probably need to change Target_Addr config param again (but otherwise it will change later).